### DISCUSSION TITLE
Preprocessing tile image data

### DISCUSSION THREAD

**Discussion by vnmiso** [fleaflet/flutter_map#1617] [Q&A] [answered]
Is it possible to preprocess downloaded raster tiles before they are shown? Specifically I want to turn them grayscale like on https://www.openrailwaymap.org
Can I create a custom `TileProvider` and/or `ImageProvider` to achieve this? If so, a basic code skeleton would be appreciated (what methods to override etc.). Thanks!

**Comment by JaffaKetchup** [comment, marked as answer]
Hey @vnmiso,
Check out the Tile Builder example in the example app. It demonstrates a dark mode, which is just a filter applied to tiles. You may be able to use the Tile Builder to achieve a satisfactory result.

**Comment by scorpiomac** [comment]
Can you please share your code with me?
For 03 weeks I've been blocked from loading the card.
I've tried Mbtiles files without success, PNG images too.

**Reply by vnmiso** [reply]
```dart
FlutterMap(
  children: [
    ColorFiltered(
      // Matrix for a grayscale filter. Found in [ColorFilter] docs
      colorFilter: const ColorFilter.matrix(<double>[
        0.2126,
        0.7152,
        0.0722,
        0,
        0,
        0.2126,
        0.7152,
        0.0722,
        0,
        0,
        0.2126,
        0.7152,
        0.0722,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
      ]),
      child: TileLayer(
        /// YOUR TILE LAYER GOES HERE
      ),
    ),
  ],          
)
```

**Reply by scorpiomac** [reply]
Thanks a lot my friend.
Is this for .mbtiles or png?
Do you have a git repo so I can see the code structure?

**Reply by vnmiso** [reply]
I just realized you might have commented on the wrong discussion and I gave you an irrelevant answer. Nevertheless, for anyone wanting a solution for the original question in this discussion, my reply is still valid.
